modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/notifications/RestServiceNotificationAndSubscriptionTestCase.java

```diff
@@ -0,0 +1,258 @@
+/*
+*Copyright (c) 2005-2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*WSO2 Inc. licenses this file to you under the Apache License,
+*Version 2.0 (the "License"); you may not use this file except
+*in compliance with the License.
+*You may obtain a copy of the License at
+*
+*http://www.apache.org/licenses/LICENSE-2.0
+*
+*Unless required by applicable law or agreed to in writing,
+*software distributed under the License is distributed on an
+*"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*KIND, either express or implied.  See the License for the
+*specific language governing permissions and limitations
+*under the License.
+*/
+
+package org.wso2.carbon.registry.es.notifications;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.wink.client.ClientResponse;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.testng.annotations.*;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.engine.frameworkutils.FrameworkPathUtil;
+import org.wso2.greg.integration.common.utils.GREGIntegrationBaseTest;
+import org.wso2.greg.integration.common.utils.GenericRestClient;
+
+import javax.ws.rs.core.MediaType;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * This class testes subscription & notification for rest services on publisher notification
+ */
+public class RestServiceNotificationAndSubscriptionTestCase extends GREGIntegrationBaseTest {
+
+    private static final Log log = LogFactory.getLog(RestServiceNotificationAndSubscriptionTestCase.class);
+
+    private TestUserMode userMode;
+    String jSessionId;
+    String assetId;
+    String cookieHeader;
+    GenericRestClient genericRestClient;
+    Map<String, String> queryParamMap;
+    Map<String, String> headerMap;
+    String publisherUrl;
+    String resourcePath;
+
+    @Factory(dataProvider = "userModeProvider")
+    public RestServiceNotificationAndSubscriptionTestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init(userMode);
+        genericRestClient = new GenericRestClient();
+        queryParamMap = new HashMap<String, String>();
+        headerMap = new HashMap<String, String>();
+        resourcePath = FrameworkPathUtil.getSystemResourceLocation()
+                + "artifacts" + File.separator + "GREG" + File.separator;
```

Comment:
Used String builder for the string appending. Refer PR https://github.com/wso2/product-greg/pull/343